docs/dev/migrating-to-v1.rst

```diff
@@ -178,8 +254,83 @@ has not been changed compared to the GridModel.
 | v0.x                         | v1                                        |
 +==============================+===========================================+
 | model.set_grid(...)          | model.grid_component.set(...)             |
++------------------------------+-------------------------------------------+
 | model.read_grid(...)         | model.grid_component.read(...)            |
++------------------------------+-------------------------------------------+
 | model.write_grid(...)        | model.grid_component.write(...)           |
++------------------------------+-------------------------------------------+
 | model.setup_grid(...)        | model.grid_component.create(...)          |
++------------------------------+-------------------------------------------+
 | model.setup_grid_from_*(...) | model.grid_component.add_data_from_*(...) |
 +------------------------------+-------------------------------------------+
+
+Changes to the format of the yaml interface
+-------------------------------------------
+
+The first change to the YAML format is that now, at the root of the documents are three keys:
+`model_type`, `global` and `steps`.
+- `model_type` details what kind of model is going to be used in the model. This used to be provided only through the CLI,
```

Comment:
Could you explain to me again why we want to offer this from the command line? Since the workflow for one type of model will generally not be compatible with the workflow of another model type (e.g. wflow yamls are not compatible with SFINCS) why would we require the user to provide redundant information?

Reply:
I don't know, clarity and user-friendliness? ``hydromt build wflow`` or ``hydromt build sfincs`` makes sure your user has some kind of idea of what he is doing. I'd like maybe to have user feedback before changing the CLI functionality too much. 
You did not change it yet in this PR anyhow so can be thought about for beta release (plugins dont really need this fast apart for updating their docs).

And I saw that ``model_type`` is also actually not implemented in the ``build`` and ``update`` methods of CLI

Reply:
Not sure I agree that it is more user-friendly but I do agree we can postpone it for now, I'll add the  comment.